lib/puppet/pops/adapters.rb

```diff
@@ -91,6 +91,7 @@ def extract_tree_text
       last = first + @adapted.length
       @adapted.eAllContents.each do |m|
         m_offset = m.offset
+        next if m_offset.nil?
```

Comment:
The expression here is "default with interpolated $::os_family" which means that the culprit TextExpression is as last as it can be. The text is still extracted correctly. I also tested with just "$::os_family". It still works. The reason is does is that offset and length of the container `ConcatenatedString` already defines the boundaries.

So unless a `TextExpression` can exist somewhere outside of a `ConcatenatingString` then fixing it doesn't give us anything besides being able to remove the check for nil. And that check cannot be removed anyway without investing in the aforementioned substantial task. So I suggest we leave things as they are.